_posts/2019-08-09-automated-prize-draws-kafka.md

```diff
@@ -0,0 +1,38 @@
+tags:       kafka, kstreams, gaming, promotions
+category:   Software Engineering
+---
+
+In the Next Generation Promotions (NGP) squad, we're always looking for new ways to be the market leader for offering great promotions. This could mean faster ways of rewarding (Choose your Cashback Hour meaning players get rewarded at the end of their time slot, rather than waiting for some overnight process to finish), better rewards (Different types of freespins, available across different game providers), or even new and improved mechanics. This post details how we've taken our old prize draw mechanic and saved our CRM and Gaming Operations teams time and effort to run them, automating something which previously would have taken hours of manual effort to figure out which customer won what and then get the prizes out to them. It also gives us the ability to run them at any time, rather than them having to start and end at midnight, and lets us give customers feedback instantly on their current ticket counts.
+
+### The Problem:
+A prize draw campaign takes up a lot of effort from multiple people in order to be run, and as such doesn't tend to be utilised as much as it could be. In addition to this the feedback from customers was negative about the use of prize draws, with many either not knowing how many tickets they have before the draw, or not really believing that prizes are awarded at all as they never hear about the results when they lose. This has a further effect on our Customer Experience team as they regularly receive calls from customers who want to know the status of the promotion and how many tickets they currently have.
+
+### The Process:
+The current main use case for prize draws is that we award a number of tickets (usually either 1 or 5 per person) as a lower tier Prize Machine prize to customers who win that day. If the customer opts in, these tickets are collated at draw time, run through Unica, and selected at random for who wins what. The most common example for Prize Machine is that 200 people will win some free spins. This can also be done based on stake if needed, with the number of tickets determined at draw time based on the amount staked by each customer.
+
+We had a number of requirements for building this mechanism in NGP; we wanted to be able to award prizes automatically at a set time, tell customers what (if anything) they've won immediately, and be able to update customers "live" with how many tickets they've got. Ideally, we'd also be able to reuse this mechanism as part of our other promotions, for example using it to replicate the experience of Bonus Time - award a random prize to a number of customers based on some criteria or other.
+
+### How we've done it:
+We split this promotion into three segments: the stake tracking (Ticket Rule Engine), the ticket storage (Wallet), and the actual draw mechanism.
+
+![Overview for the three components that make up a Prize Draw](/images/prize-draw/overview.png)
```

Comment:
I think it's worth splitting this out into three separate images, as it's kinda small and difficult to read on the page.

See live preview: https://kafka-prize-draw--lmhd-skybet.netlify.com/2019/08/09/automated-prize-draws-kafka/